core_v2/sources/v2_domains.move

```diff
@@ -0,0 +1,1318 @@
+        // Else, create a new one and transfer it to the signer
+        let account_addr = signer::address_of(sign);
+        let token_addr = get_token_addr_inline(domain_name, subdomain_name);
+        if (object::is_object(token_addr)) {
+            let record = borrow_global_mut<NameRecord>(token_addr);
+            record.expiration_time_sec = name_expiration_time_secs;
+            record.target_address = option::none();
+            record.registration_time_sec = timestamp::now_seconds();
+            object::transfer_with_ref(object::generate_linear_transfer_ref(&record.transfer_ref), account_addr);
+        } else {
+            create_token(
+                account_addr,
+                domain_name,
+                subdomain_name,
+                name_expiration_time_secs,
+            );
+        };
+
+        event::emit_event<RegisterNameEvent>(
+            &mut borrow_global_mut<RegisterNameEvents>(@aptos_names_v2).register_name_events,
+            RegisterNameEvent {
+                domain_name,
+                subdomain_name,
+                registration_fee_octas: price,
+                expiration_time_secs: name_expiration_time_secs,
+            },
+        );
+    }
+
+    // === RENEW DOMAIN ===
+
+    public fun renew_domain(
+        sign: &signer,
+        domain_name: String,
+        renewal_duration_secs: u64,
+    ) acquires CollectionCapability, NameRecord, SubdomainExt, RenewNameEvents, ReverseRecord {
+        // check the domain eligibility
+        let length = validate_name_string(domain_name);
+
+        validate_registration_duration(renewal_duration_secs);
+        assert!(is_domain_in_renewal_window(domain_name), error::invalid_state(EDOMAIN_NOT_AVAILABLE_TO_RENEW));
+        let price = v2_price_model::price_for_domain(length, renewal_duration_secs);
+        // pay the price
+        coin::transfer<AptosCoin>(sign, v2_config::fund_destination_address(), price);
+        renew_domain_internal(domain_name, renewal_duration_secs, price);
+    }
+
+    fun renew_domain_internal(
+        domain_name: String,
+        renewal_duration_secs: u64,
+        price: u64,
+    ) acquires CollectionCapability, NameRecord, RenewNameEvents, ReverseRecord {
+        let record = get_record_mut(domain_name, option::none());
+        record.expiration_time_sec = record.expiration_time_sec + renewal_duration_secs;
+
+        // Idea here is that if this is a primary name, then the target_addr's reverse lookup should point back to this domain
+        let is_primary_name = if (option::is_some(&record.target_address)) {
+            let maybe_reverse_record = if (exists<ReverseRecord>(*option::borrow(&record.target_address))) {
+                let reverse_record = borrow_global<ReverseRecord>(*option::borrow(&record.target_address));
+                reverse_record.token_addr
+            } else {
+                option::none()
+            };
+            if (option::is_some(&maybe_reverse_record)) {
```

Comment:
Why not combine this if with the previous one (if (exists<ReverseRecord>(*option::borrow(&record.target_address))))? Seems unnecessary to break this into 2 separate conditionals and returning the option in between if none of the intermediate results is used outside

Reply:
Can't it be:

if (exists<ReverseRecord>(*option::borrow(&record.target_address))) {
     let reverse_record = borrow_global<ReverseRecord>(*option::borrow(&record.target_address));
     get_token_addr_inline(domain_name, option::none()) == reverse_record.token_addr
} else {
  false
}


?

Reply:
yeah, I agree with Kevin this isn't ... that complex of a nest